src/Chainweb/Pact/Backend/RelationalCheckpointer.hs

```diff
@@ -152,6 +154,7 @@ doSave dbenv hash = runBlockEnv dbenv $ do
     -- FIXME: if any of the above fails with an exception the following isn't
     -- executed and a pending SAVEPOINT is left on the stack.
     commitSavepoint Block
+    checkNotInTx logger
```

Comment:
I think, this can trigger false positives as well. Although not every call of `withCheckpointer` calls `_cpSave`, I think, the decision is independent from wether the call is nested or not:

```.haskell
        try (restore (act cenv)) >>= \case
            Left !e -> discardTx checkPointer >> throwM @_ @SomeException e
            Right (Discard !result) -> discardTx checkPointer >> return result
            Right (Save header !result) -> saveTx checkPointer header >> return result
```